lib/plugins/aws/deploy/lib/cleanupS3Bucket.js

```diff
@@ -12,18 +10,19 @@ module.exports = {
       5
     );
 
-    const service = this.serverless.service.service;
-    const stage = this.provider.getStage();
-    const prefix = this.provider.getDeploymentPrefix();
-
-    const response = await this.provider.request('S3', 'listObjectsV2', {
-      Bucket: this.bucketName,
-      Prefix: `${prefix}/${service}/${stage}`,
-    });
-    const stacks = findAndGroupDeployments(response, prefix, service, stage);
+    const stacks = await this.findDeployments();
```

Comment:
> Are there any statistics that would confirm that the affected functionality (cleanup S3 bucket, rollback, deploy list) has to be optimized? 

I believe that we need to change how things works only for _cleanup_ operation, and there are no changes to _rollback_ and _deploy list_ planned per spec. Do you see that some are needed? If so can you please elaborate in an issue, as that definitely needs to be included in a spec then.

Now when speaking strictly about _cleanup_, this change will introduce a need to download CF templates as hosted on S3. In case of large services those templates can be big. I assume that downloading those templates will be relatively fast when comparing to the deploy operation in general, but I suspect that in case of large services it still may add 0.1-0.5s of overhead to the command, which we can avoid in some scenarios.

Also do you see anything particularly challenging in what I proposed? My feeling is that this doesn't make our implementation more complex, but rather better organized.